clang/include/clang/Basic/DiagnosticFrontendKinds.td

```diff
@@ -384,4 +384,8 @@ def warn_hlsl_langstd_minimal :
   Warning<"support for HLSL language version %0 is incomplete, "
           "recommend using %1 instead">,
   InGroup<HLSLDXCCompat>;
+
+def warn_name_cannot_be_demangled : Warning<
+  "cannot demangle the name '%0'">,
+  InGroup<CXX20Compat>;
```

Comment:
Group is wrong here, we likely want to give it its own group for the diagnostic.

Reply:
Added a new `Demangler` group.